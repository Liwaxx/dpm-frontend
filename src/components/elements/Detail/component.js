import React from 'react';
import PropTypes from 'prop-types';
import Dialog from '../Dialog';
import Minibox from '../MiniBox';
import PaperGrid from '../PaperGrid';
import Text from '../Text';

export default class Component extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total1: '',
      status1: '',
      task1: [],
      open: true,
    };
  }

  handleClickOpen = scroll => () => {
    this.setState({ 'open': true, scroll });
  };

  handleClose = () => {
    this.setState({ 'open': false });
  };

  render() {
    const { classes, detail } = this.props;
    const backlog_jumlah = '1000';

    return (
      <Dialog>
        <section className={classes.container}>
          <div>
            <div className={classes.search}>
              <Text className={classes.title_total_backlog}>
                Total Backlog <b>{backlog_jumlah}</b>
              </Text>
              <div className={classes.search2}>
                <input type="text" className={classes.searchEl2} />
                <button className={classes.searchEl}>Search</button>
              </div>
            </div>
            <br />
          </div>

          {
            detail.totalBacklog ? (
              detail.totalBacklog.map((item, index) => (
                <div className={classes.first} key={index}>
                  <Text className={classes.keterangan}>
                    {item.status}
                  </Text>
                  <Text className={classes.keterangan1}>
                    {item.total.toString()}
                  </Text>
                  <br/>
                  <PaperGrid>
                    <div className={classes.pembuka}>
                      {item.description.map((value, index) => {
                        return (<Minibox key={index}
                          classes={classes}
                          title={value.description}
                          description={'CJ-245'} />);
                      })}
                    </div>
                  </PaperGrid>
                </div>
              ))
            ) : ''
          }

          <div className={classes.first}>
            <Text className={classes.keterangan}>{this.state.status4}</Text>
            <Text className={classes.keterangan1}>{this.state.total4}</Text>

            <PaperGrid>
              <div className={classes.pembuka}>
                {this.state.task1.map((value, index) => {
                  return (<Minibox key={index}
                    classes={classes}
                    title={value.status}
                    description={value.total} />);
                })}
              </div>
            </PaperGrid>
          </div>

          <div className={classes.second}>
            <Text className={classes.keterangan}>{this.state.status2}</Text>
            <Text className={classes.keterangan1}>{this.state.total2}</Text>
        
            <PaperGrid>
              <div className={classes.pembuka}>
                {this.state.task1.map((value, index) => {
                  return (<Minibox key={index}
                    classes={classes}
                    title={value.status}
                    description={value.total} />);
                })}
              </div>
            </PaperGrid>
          </div>
        </section>
      </Dialog>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object,
  actions: PropTypes.object,
  isLoading: PropTypes.bool,
  detail: PropTypes.object
};