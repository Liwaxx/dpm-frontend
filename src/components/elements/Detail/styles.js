const styles = {
  pembuka: {
    backgroundColor: '#EBEFF5',
    width: '100%',
    height: '100%',
  },
  title_total_backlog: {
    textAlign: 'left',
    fontSize: 15,
    float: 'left',
    width: '80%',
    marginBottom: 10
  },
  search: {
    width: '100%'
  },
  searchEl: {
    backgroundColor: '#EBEFF5',
    outline: 'white',
  },
  search2: {
    float: 'right',
    width: '20%',
    marginBottom: 10
  },
  container: {
    textAlign: 'center',
    margin: 0,
    padding: 15,
    paddingTop: 0,
    backgroundColor: 'white',
    minHeight: '100vh',
  },
  keterangan: {
    textAlign: 'left',
    float: 'left',
    margin: 0,
    paddingTop: 10,
    paddingLeft: 10,
    backgroundColor: '#EBEFF5',
    width: '75%',
    marginBottom: 10
  },
  keterangan1: {
    paddingTop: 10,
    paddingRight: 10,
    float: 'right',
    textAlign: 'right',
    margin: 0,
    backgroundColor: '#EBEFF5',
    width: '25%',
  },
  first: {
    width: '19%',
    float: 'left',
    marginRight: '1%'
  },
  second: {
    width: '20%',
    float: 'left',
    marginRight: '0'
  },
  paper: {
    margin: 'auto',
    paddingBottom: 10,
    height: '100%',
    textAlign: 'left',
    fontSize: 13
  },
  code: {
    color: '#BCC3CD',
  },
  modal: {
    width: '80%',
    margin: 'auto'
  },
};

export default styles;