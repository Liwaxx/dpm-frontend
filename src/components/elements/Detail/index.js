import { withStyles } from '@material-ui/core/styles';
import Detail from './component';
import styles from './styles';

export default withStyles(styles)(Detail);