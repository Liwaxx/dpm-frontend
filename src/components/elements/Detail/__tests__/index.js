import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '../index';

jest.mock('../../Dialog');
jest.mock('../../MiniBox');
jest.mock('../../PaperGrid');
jest.mock('../../Text');

describe('Detail', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Detail detail={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with data', () => {
    const detail = {
      totalBacklog: [{
        status: 'tes',
        total: 1,
        description: [{ description: 'tes', }],
      }]
    };
    const tree = renderer
      .create(<Detail detail={detail} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});