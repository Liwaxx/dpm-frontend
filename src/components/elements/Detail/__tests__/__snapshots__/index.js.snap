// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail renders correctly 1`] = `null`;

exports[`Detail renders with data 1`] = `null`;
